test/e2e/tcp/connectivity.go

```diff
@@ -0,0 +1,116 @@
+package tcp
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/submariner-io/shipyard/test/e2e/framework"
+)
+
+type ConnectivityTestParams struct {
+	Framework             *framework.Framework
+	UseService            bool
+	Networking            framework.NetworkingType
+	ConnectionTimeout     uint
+	ConnectionAttempts    uint
+	FromCluster           framework.ClusterIndex
+	FromClusterScheduling framework.NetworkPodScheduling
+	ToCluster             framework.ClusterIndex
+	ToClusterScheduling   framework.NetworkPodScheduling
+}
+
+func RunConnectivityTest(p ConnectivityTestParams) (*framework.NetworkPod, *framework.NetworkPod) {
+	if p.ConnectionTimeout == 0 {
+		p.ConnectionTimeout = framework.TestContext.ConnectionTimeout
+	}
+
+	if p.ConnectionAttempts == 0 {
+		p.ConnectionAttempts = framework.TestContext.ConnectionAttempts
+	}
+
+	listenerPod, connectorPod := createPods(&p)
+	listenerPod.CheckSuccessfulFinish()
+	connectorPod.CheckSuccessfulFinish()
+
+	By("Verifying that the listener got the connector's data and the connector got the listener's data")
+	Expect(listenerPod.TerminationMessage).To(ContainSubstring(connectorPod.Config.Data))
+	Expect(connectorPod.TerminationMessage).To(ContainSubstring(listenerPod.Config.Data))
+
+	// Normally, Submariner preserves the sourceIP for inter-cluster traffic. However, when a POD is using the
+	// HostNetwork, it does not get an IPAddress from the podCIDR and it uses the HostIP. Submariner, for such PODs,
+	// would MASQUERADE the external traffic from that POD to the corresponding CNI interface ip-address on that
+	// Host. So, we skip source-ip validation for PODs using HostNetworking.
+	if p.Networking == framework.PodNetworking {
+		By("Verifying the output of listener pod which must contain the source IP")
+		Expect(listenerPod.TerminationMessage).To(ContainSubstring(connectorPod.Pod.Status.PodIP))
+	}
+
+	// Return the pods in case further verification is needed
+	return listenerPod, connectorPod
+}
+
+func RunNoConnectivityTest(p ConnectivityTestParams) (*framework.NetworkPod, *framework.NetworkPod) {
+	if p.ConnectionTimeout == 0 {
+		p.ConnectionTimeout = 5
```

Comment:
I guess this is the code we had, but while we are at refactors I don't want us to forget about it :)

Such timeout is too low:
1) we create listener with 5 sec timeout,
2) we go to 2nd cluster, and create connector, with the same timeout

but chances are that the creation time will be much higher than the connection attempts.

This leads to false positives, and we should probably bump this to at least (arbitrary) 30 seconds. I know it will make this test slower, but it will have less unlike false possitives.

We could refine our testing images in the future for better handling of timeouts, etc.

Reply:
yeah this may be too low but this PR is to simply port the code to shipyard - I'd prefer such a change to be done in a separate PR.

Reply:
Makes sense, I converted it to an issue: https://github.com/submariner-io/shipyard/issues/40